airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -197,31 +206,55 @@ private void produceMessages(final AutoCloseableIterator<AirbyteMessage> message
   }
 
   @VisibleForTesting
-  static void consumeWriteStream(final AirbyteMessageConsumer consumer) throws Exception {
-    // use a Scanner that only processes new line characters to strictly abide with the
-    // https://jsonlines.org/ standard
-    final Scanner input = new Scanner(System.in, StandardCharsets.UTF_8).useDelimiter("[\r\n]+");
-    consumer.start();
-    while (input.hasNext()) {
-      consumeMessage(consumer, input.next());
+  static void consumeWriteStream(final SerializedAirbyteMessageConsumer consumer) throws Exception {
+    try (final BufferedInputStream bis = new BufferedInputStream(System.in);
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+      consumeWriteStream(consumer, bis, baos);
     }
   }
 
-  private static void runConsumer(final AirbyteMessageConsumer consumer) throws Exception {
-    watchForOrphanThreads(
-        () -> consumeWriteStream(consumer),
-        () -> System.exit(FORCED_EXIT_CODE),
-        INTERRUPT_THREAD_DELAY_MINUTES,
-        TimeUnit.MINUTES,
-        EXIT_THREAD_DELAY_MINUTES,
-        TimeUnit.MINUTES);
+  @VisibleForTesting
+  static void consumeWriteStream(final SerializedAirbyteMessageConsumer consumer,
+                                 final BufferedInputStream bis,
+                                 final ByteArrayOutputStream baos)
+      throws Exception {
+    consumer.start();
+
+    final byte[] buffer = new byte[8192]; // 8K buffer
+    int bytesRead;
+    int byteCount = 0;
+    boolean lastWasNewLine = false;
+
+    while ((bytesRead = bis.read(buffer)) != -1) {
+      for (int i = 0; i < bytesRead; i++) {
+        final byte b = buffer[i];
+        if (b == '\n' || b == '\r') {
```

Comment:
[nit] Any worries about "encoded" newline characters within the records themselves (`{body: "hello\nworld"}`)?  Is there a builtin for `sytem.newline` or something like that?

Reply:
I have a test in `IntegrationRunnerBackwardsCompatbilityTest.java` to make sure the behavior is the same between what we have on master and this branch to make sure we don't mess anything up.